pkg/scaler/utils.go

```diff
@@ -57,3 +60,36 @@ func isRateLimited(err error) bool {
 	const prefix = "rate: "
 	return strings.HasPrefix(err.Error(), prefix)
 }
+
+func shootHibernationStateChanged(old, new *gardenerv1alpha1.Cluster) bool {
+	decoder, err := extensionscontroller.NewGardenDecoder()
+	if err != nil {
+		klog.V(4).Infof("Error getting gardener decoder for cluster %v. Err: %v", new.Name, err)
+		return false
+	}
+
+	oldShoot, err := extensionscontroller.ShootFromCluster(decoder, old)
+	if err != nil {
+		klog.V(4).Infof("Error getting old shoot version from cluster: %v. Err: %v", old.Name, err)
+		return false
+	}
+	newShoot, err := extensionscontroller.ShootFromCluster(decoder, new)
+	if err != nil {
+		klog.V(4).Infof("Error getting new shoot version from cluster: %v. Err: %v", new.Name, err)
+		return false
+	}
+
+	if oldShoot.Status.IsHibernated != newShoot.Status.IsHibernated {
+		return true
+	}
+
+	if (oldShoot.Spec.Hibernation == nil && newShoot.Spec.Hibernation != nil) ||
```

Comment:
Can't we use `reflect.DeepEqual` here?

Reply:
just wanted to avoid updates to other fields in the `Hibernation` field.
It can look like this:
```yaml
hibernation:
    enabled: false
    schedules:
      - start: '00 20 * * 1,2,3,4,5,6,0'
        end: '30 08 * * 1,2,3,4,5,6,0'
        location: Europe/Berlin
      - start: '30 05 * * 1,2,3,4,5,6,0'
        end: '30 01 * * 1,2,3,4,5,6,0'
        location: Europe/Berlin
```
wdyt?